File: js/examples/time_hash_example.ts
import { Hash, newWithTime, hashWithTime } from '../src/hash';

// Example of using time-based hash functions
console.log('TOPAY-Z512 Time-based Hash Example\n');

// Create a hash using the current time with the static method
const timeHash = Hash.newWithTime();
console.log(`Time-based hash: ${timeHash.toHex()}`);

// Create another hash after a short delay
setTimeout(() => {
  const timeHash2 = Hash.newWithTime();
  console.log(`Time-based hash after 1 second: ${timeHash2.toHex()}`);
  
  // Demonstrate that the hashes are different
  console.log(`Hashes are different: ${!timeHash.equals(timeHash2)}`);
  
  // Using the convenience functions
  const timeHashObj = newWithTime();
  console.log(`Time-based hash object: ${timeHashObj.toHex()}`);
  
  const timeHashBytes = hashWithTime();
  console.log(`Time-based hash bytes length: ${timeHashBytes.length}`);
}, 1000);

File: go/examples/time_hash_example.go
// TOPAY-Z512 Time-based Hash Example
//
// This example demonstrates how to generate hashes using the current time
// with the TOPAY-Z512 library.
package examples

import (
	"fmt"
	"time"

	"github.com/topay-foundation/topayz512/pkg/hash"
)

// RunTimeHashExample demonstrates time-based hash generation
func RunTimeHashExample() {
	fmt.Println("TOPAY-Z512 Time-based Hash Example")

	// Create a hash using the current time
	timeHash := hash.NewWithTime()
	fmt.Printf("Time-based hash: %s\n", timeHash)

	// Create another hash after a short delay
	time.Sleep(1 * time.Second)
	timeHash2 := hash.NewWithTime()
	fmt.Printf("Time-based hash after 1 second: %s\n", timeHash2)

	// Demonstrate that the hashes are different
	fmt.Printf("Hashes are different: %v\n", timeHash != timeHash2)

	// Using the convenience function
	timeHashBytes := hash.Sum512WithTime()
	fmt.Printf("Time-based hash bytes: %v\n", timeHashBytes)
	fmt.Printf("Time-based hash bytes length: %d\n", len(timeHashBytes))
}